pkg/gitreceive/build.go

```diff
@@ -183,14 +183,19 @@ func build(
 		return fmt.Errorf("creating builder pod (%s)", err)
 	}
 
-	if err := waitForPod(kubeClient, newPod.Namespace, newPod.Name, conf.SessionIdleInterval(), conf.BuilderPodTickDuration(), conf.BuilderPodWaitDuration()); err != nil {
+	pw := k8s.NewPodWatcher(kubeClient, "deis")
```

Comment:
That's when you see the benefit. Please try to build more than one application at the same time (checking the amount of packets against the api server)